package.json

```diff
@@ -19,6 +19,7 @@
   "dependencies": {
     "@types/lodash": "^4.14.136",
     "@types/ws": "^6.0.3",
+    "backo": "^1.1.0",
```

Comment:
I think this is only ~50 LOC. Instead of pulling in a new dependency here, can we just handle this logic internally?

If we want to keep `src/common/connection.ts` focused, this can be in a separate file.

Reply:
fine by me! A little extra work, but it's an old package so re-writing it in TypeScript will be good for maintenance.